java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -757,6 +760,79 @@ public static Table concatenate(Table... tables) {
     return new Table(concatenate(tableHandles));
   }
 
+  /**
+   * Given a sorted inputTable return the lower bound.
+   * Example:
+   *
+   *  Single column:
+   *      idx            0   1   2   3   4
+   *   inputTable  =   { 10, 20, 20, 30, 50 }
+   *   valuesTable =   { 20 }
+   *   result      =   { 1 }
+   *
+   *  Multi Column:
+   *      idx                0    1    2    3    4
+   *   inputTable      = {{  10,  20,  20,  20,  20 },
+   *                      { 5.0,  .5,  .5,  .7,  .7 },
+   *                      {  90,  77,  78,  61,  61 }}
+   *   valuesTable     = {{ 20 },
+   *                      { .7 },
+   *                      { 61 }}
+   *   result          = {  3 }
+   * NaNs in column values do not work especially
+   * if the table has multiple columns.
+   * Strings and String categories do not work for this method. If the input table is
+   * unsorted the results are wrong. Types of columns can be of mixed data types and
+   * the result gives the index of the value pertaining to the matching type column
+   * in the input table.
+   * @param inputTable the sorted table as input for bounds
+   * @param valueTable the table of values that need to be inserted
+   * @param descFlags indicates the ordering of the column(s), true if descending
+   * @param nullsAsLargest true if nulls are assumed largest
+   * @return ColumnVector with lower bound indices for all rows in valueTable
+   */
+  public ColumnVector lowerBound(Table inputTable, Table valueTable, boolean[] descFlags,
+                                 boolean nullsAsLargest) {
+    return new ColumnVector(gdfBound(inputTable.nativeHandle, valueTable.nativeHandle,
```

Comment:
nit: if we have preconditions for the function we should make sure that either the native code handles throwing a readable exception or we do.  At a minimum with an assert.

Does this handle mismatched types for `inputTable` and `valueTable`?

Reply:
> Does this handle mismatched types for inputTable and valueTable?
It does not seem to be doing that and returns unexpected int values for indices. Will add check for that as well and probably best to open a bug report for this. Thanks for pointing it out.